src/Math-Matrix/PMTensor.class.st

```diff
@@ -0,0 +1,145 @@
+Class {
+	#name : #PMTensor,
+	#superclass : #Object,
+	#instVars : [
+		'array',
+		'shape',
+		'first',
+		'strides'
+	],
+	#category : #'Math-Matrix'
+}
+
+{ #category : #'instance creation' }
+PMTensor class >> newWith: anInteger [
+
+	^ self new array: {anInteger} withShape: #(  )
+]
+
+{ #category : #'instance creation' }
+PMTensor class >> rows: anArray [
+
+	^ self new array: anArray flattened withShape: (self shape:anArray)
+]
+
+{ #category : #accessing }
+PMTensor class >> shape: anArray [
+
+anArray isArray ifFalse:[^#()]
+ifTrue:[
+	^ {anArray size}, (self shape: anArray first)
+	]
+
+]
+
+{ #category : #comparing }
+PMTensor >> = aTensor [
+
+	^ array = aTensor array & (first = aTensor first)
+	  & (strides = aTensor strides) & (shape = aTensor shape)
+]
+
+{ #category : #accessing }
+PMTensor >> array [ 
+^array
+]
+
+{ #category : #'as yet unclassified' }
+PMTensor >> array: aFlatArray withShape: aShape [
+
+	array := aFlatArray.
+	shape := aShape copy.
+	self updateFirst.
+	shape ifNotEmpty: [ self updateStrides]
+]
+
+{ #category : #accessing }
+PMTensor >> first [ 
+^first
+]
+
+{ #category : #public }
+PMTensor >> get: coords [
+
+	| position |
+	position := self getPosition: coords.
+	^ array at: position
+]
+
+{ #category : #'primitives - file' }
+PMTensor >> getPosition: coords [
```

Comment:
From the test, this could be elementAt

Reply:
The tests are misleading. The return value is the index into the  flattened  version.

suggestion: flattenedIndexOf: